src/argilla/client/models.py

```diff
@@ -37,16 +36,27 @@ class _Validators(BaseModel):
     """Base class for our record models that takes care of general validations"""
 
     @validator("metadata", check_fields=False)
-    def _check_value_length(cls, v):
-        """Checks metadata values length and apply value truncation for large values"""
-        new_metadata = limit_value_length(v, max_length=MAX_KEYWORD_LENGTH)
-        if new_metadata != v:
+    def _check_value_length(cls, metadata):
+        """Checks metadata values length and warn message for large values"""
+        if not metadata:
+            return metadata
+
+        default_length_exceeded = False
+        for v in metadata.values():
+            if isinstance(v, str) and len(v) > DEFAULT_MAX_KEYWORD_LENGTH:
+                default_length_exceeded = True
+                break
+
+        if default_length_exceeded:
             warnings.warn(
-                "Some metadata values exceed the max length. Those values will be"
-                f" truncated by keeping only the last {MAX_KEYWORD_LENGTH} characters."
+                "Some metadata values could exceed the max length. For those cases, values will be"
+                f" truncated by keeping only the last {DEFAULT_MAX_KEYWORD_LENGTH} characters. "
+                "You can configure setup this length in the server with the ARGILLA_METADATA_FIELD_LENGTH"
```

Comment:
You can configure this length in the server with the ARGILLA_METADATA_FIELD_LENGTH environment variable. Note that, setting this too high may lead to Elastic performance issues.

Reply:
done